### ISSUE TITLE
Debug assistance requested for 'undefined' Zigbee model for Aqara SSM-U01

### ISSUE THREAD

**Issue by sunsear** [Koenkk/zigbee-herdsman-converters#2485]
After adding a ton of devices successfully, my first [Aqara SSM-U01](https://www.zigbee2mqtt.io/devices/SSM-U01.html) simply doesn't identify itself properly. I found several other people with issues with Xiaomi devices in (https://github.com/Koenkk/zigbee2mqtt/issues/1378), but that issue was apparently resolved. Somehow I cannot get this device to pair correctly, even after reading the FAQ, tons of retries, restarts and factory resets.

Can somebody help me debug what is going on here?

The device pairing produces the following logging:

```
Debug Device '0x54ef44100009dda8' announced itself
Info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x54ef44100009dda8","ieee_address":"0x54ef44100009dda8"},"type":"device_announce"}'
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x54ef44100009dda8"},"type":"device_announced"}'
Debug Device '0x54ef44100009dda8' announced itself
Info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x54ef44100009dda8","ieee_address":"0x54ef44100009dda8"},"type":"device_announce"}'
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x54ef44100009dda8"},"type":"device_announced"}'
Debug Received Zigbee message from '0x54ef44100009dda8', type 'readResponse', cluster 'genBasic', data '{"manufacturerName":"LUMI"}' from endpoint 1 with groupID 0
Debug Skipping message, definition is undefined and still interviewing
Debug Received Zigbee message from '0x54ef44100009dda8', type 'readResponse', cluster 'genBasic', data '{"powerSource":0}' from endpoint 1 with groupID 0
Debug Skipping message, definition is undefined and still interviewing
Debug Received Zigbee message from '0x54ef44100009dda8', type 'readResponse', cluster 'genBasic', data '{"zclVersion":3}' from endpoint 1 with groupID 0
Debug Skipping message, definition is undefined and still interviewing
Debug Received Zigbee message from '0x54ef44100009dda8', type 'readResponse', cluster 'genBasic', data '{"appVersion":21}' from endpoint 1 with groupID 0
Debug Skipping message, definition is undefined and still interviewing
Debug Received Zigbee message from '0x54ef44100009dda8', type 'readResponse', cluster 'genBasic', data '{"stackVersion":0}' from endpoint 1 with groupID 0
Debug Skipping message, definition is undefined and still interviewing
Debug Received Zigbee message from '0x54ef44100009dda8', type 'readResponse', cluster 'genBasic', data '{"hwVersion":1}' from endpoint 1 with groupID 0
Debug Skipping message, definition is undefined and still interviewing
Debug Received Zigbee message from '0x54ef44100009dda8', type 'readResponse', cluster 'genBasic', data '{"dateCode":"Aug 8 2020"}' from endpoint 1 with groupID 0
Debug Skipping message, definition is undefined and still interviewing
Debug Received Zigbee message from '0x54ef44100009dda8', type 'readResponse', cluster 'genBasic', data '{"swBuildId":""}' from endpoint 1 with groupID 0
Debug Skipping message, definition is undefined and still interviewing
Info Successfully interviewed '0x54ef44100009dda8', device has successfully been paired
Warning Device '0x54ef44100009dda8' with Zigbee model 'undefined' and manufacturer name 'LUMI' is NOT supported, please follow https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html
Info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"definition":null,"friendly_name":"0x54ef44100009dda8","ieee_address":"0x54ef44100009dda8","status":"successful","supported":false},"type":"device_interview"}'
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_successful","meta":{"friendly_name":"0x54ef44100009dda8","supported":false},"type":"pairing"}'
```

**Comment by Koenkk** [comment]
Try force removing the device and pair it closely to the coordinator (then move it to the desired location): https://www.zigbee2mqtt.io/information/FAQ.html#interview-fails

**Comment by sunsear** [comment]
Unfortunately I have done that both already and retried it again. Shall I put some herdsman logging up in an issue?

**Comment by Koenkk** [comment]
Yes please post the log (on pastebin.com and share the link here)

**Comment by sunsear** [comment]
All right, I took the following steps: 
1. disabled join permission
2. force removed the SSM-u01
3. powered down the SSM-u01
4. Enabled permission to join
5. factory reset the SSM-u01 by keeping the button pressed for roughly 5 seconds
6. got the herdsman logging: 

Below the zigbee2m

**Comment by Koenkk** [comment]
Can you try replacing `/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/controller/model/device.js` with https://gist.github.com/Koenkk/b78a22f0b1cb4a520b7e307e491da501 and provide the herdsman debug logging again? (does it get a modelID now?)

**Comment by sunsear** [comment]
Yes, I would love to try and do that, but I am running from HassOS and have already tried changing device.js for an evening. I will try again this evening, would anyone happen to have a pointer on this? 

The only thing I think I can try is to create a volume where I put the herdsman converters and load that up somehow. The supervisor however does not seem to appreciate it when I change options like volumes from Portainer. I'll give it a go again anyways.

**Comment by Koenkk** [comment]
You should be able to SSH into hassos, `docker exec -it CONTAINER_ID sh` to get into the container, execute `nano node_modules/zigbee-herdsman/dist/controller/model/device.js`, remove everything and update with the link provided, save and exit. Exit the docker container and restart it with `docker restart CONTAINER_ID`.

**Comment by sunsear** [comment]
I thought I tried exactly that and that my changes got overwritten. You recipe worked on both ends, I got in, managed to change device.js and now have a working Aqara SSM-U01. Great stuff!

Now I'm curious, going to check your diff. I love open source ;-)

**Comment by sunsear** [comment]
Wow, I still have no idea how Zigbee works, but it looks like you're retrying to read the model id again after 10 seconds? Sounds a bit long, but glad this worked. I am indeed seeing that log line in my herdsman logging. 

Small tip, there's a typo in:

debug.log(`Interview - first modelID attempt failed, retryting..`);

I cannot see a branch to submit a PR to, so here's a proposed fix: 

debug.log(`Interview - first modelID retrieval attempt failed, retrying after 10 seconds...`);

**Comment by Koenkk** [comment]
Thanks, integrated the fix (https://github.com/Koenkk/zigbee-herdsman/commit/f2d196ab3e2c32c802ec45afa4420ed55f304ff7)

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

**Comment by jd1900** [comment]
Could this be related to this issue I have?
https://github.com/Koenkk/zigbee2mqtt/issues/7942

**Comment by jd1900** [comment]
> Thanks, integrated the fix ([Koenkk/zigbee-herdsman@f2d196a](https://github.com/Koenkk/zigbee-herdsman/commit/f2d196ab3e2c32c802ec45afa4420ed55f304ff7))
> 
> Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

Could this be related to this issue I have?
https://github.com/Koenkk/zigbee2mqtt/issues/7942
